rdflib/plugins/stores/sparqlconnector.py

```diff
@@ -94,47 +92,44 @@ def query(self, query, default_graph=None):
 
         if self.method == "GET":
             args["params"].update(params)
+            qsa = "?" + urlencode(args["params"])
+            res = urlopen(Request(self.query_endpoint + qsa, headers=args["headers"]))
         elif self.method == "POST":
             args["headers"].update({"Content-Type": "application/sparql-query"})
-            args["data"] = params
```

Comment:
Why don't we need `params` in `args["data"]` anymore?

Reply:
The only `params` allowed for a `POST` SPARQL query is the query itself, [as per the spec](https://www.w3.org/TR/sparql11-protocol/#query-operation), so we don't allow any others. Auth params for secure endpoints I've catered for with a separate `auth` param.

Reply:
We still need to submit `default-graph-uri` and `named-graph-uri` parameters. So the `params` is still required.

Also I think we should support both `application/x-www-form-urlencoded` and `application/sparql-query` content-types. But I'm unsure how we would determine which content-type is supported by the store we are speaking with.

Reply:
The `params` attribute is still required or at least the information about the default-graph.

If you consider the following:

Load some context aware quad store (e.g. the QuitStore) with the following data:
```turtle
graph <http://example.org/> {
    <http://example.org/ExampleInstance> a <http://example.org/Example>
}
graph <http://othergraph.org/> {
    <http://example.org/OtherInstance> a <http://example.org/Example>
}
```

The query endpoint is assumed to be available at `http://localhost:5000/sparql`

```python
class SPARQLStoreQuitStoreTestCase(unittest.TestCase):
    store_name = "SPARQLStore"
    path = "http://localhost:5000/sparql"
    create = False

    def setUp(self):
        store = SPARQLStore(query_endpoint=self.path, method="POST")
        self.conjunctivegraph = ConjunctiveGraph(store=store)

    def tearDown(self):
        self.conjunctivegraph.close()

    def test_Query(self):
        query = "select distinct ?inst where {?inst a <http://example.org/Example>}"
        graph = self.conjunctivegraph.get_context(URIRef("http://example.org/"))
        res = graph.query(query, initNs={})
        assert len(res) == 1, len(res)
        for i in res:
            assert type(i[0]) == URIRef, i[0].n3()
            assert i[0] == URIRef("http://example.org/ExampleInstance"), i[0].n3()
```

The query is executed as POST request but does not convey the information about the default graph.

This issue is a combination of removing this line and sending the wrong Content-Type.
Actually, what the request was doing before this change and the merge of #1022 was sending a proper POST request with a `application/x-www-form-urlencoded` content type according to the SPARQL 1.1 Protocol (query via URL-encoded POST; https://www.w3.org/TR/2013/REC-sparql11-protocol-20130321/#query-operation).